Common/Data/Consolidators/VolumeRenkoConsolidator.cs

```diff
@@ -0,0 +1,289 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Data.Consolidators
+{
+    /// <summary>
+    /// This consolidator can transform a stream of <see cref="BaseData"/> instances into a stream of <see cref="RenkoBar"/>
+    /// with a constant volume for each bar.
+    /// </summary>
+    public class VolumeRenkoConsolidator : IDataConsolidator
+    {
+        private bool _firstTick = true;
+        private DataConsolidatedHandler _dataConsolidatedHandler;
+        private VolumeRenkoBar _currentBar;
+        private IBaseData _consolidated;
+        private decimal _volumeLeftOver = 0m;
+        
+        /// <summary>
+        /// Time of consolidated close.
+        /// </summary>
+        /// <remarks>Protected for testing</remarks>
+        protected DateTime CloseOn;
+
+        /// <summary>
+        /// Value of consolidated close.
+        /// </summary>
+        /// <remarks>Protected for testing</remarks>
+        protected decimal CloseRate;
+
+        /// <summary>
+        /// Value of consolidated high.
+        /// </summary>
+        /// <remarks>Protected for testing</remarks>
+        protected decimal HighRate;
+
+        /// <summary>
+        /// Value of consolidated low.
+        /// </summary>
+        /// <remarks>Protected for testing</remarks>
+        protected decimal LowRate;
+
+        /// <summary>
+        /// Time of consolidated open.
+        /// </summary>
+        /// <remarks>Protected for testing</remarks>
+        protected DateTime OpenOn;
+
+        /// <summary>
+        /// Value of consolidate open.
+        /// </summary>
+        /// <remarks>Protected for testing</remarks>
+        protected decimal OpenRate;
+
+        /// <summary>
+        /// Size of the consolidated bar.
+        /// </summary>
+        /// <remarks>Protected for testing</remarks>
+        protected decimal BarSize;
+
+        /// <summary>
+        /// Gets the kind of the bar
+        /// </summary>
+        public RenkoType Type => RenkoType.Wicked;
```

Comment:
> public RenkoType Type

Don't think it's a good idea to use `Type` as name for variable. Instead of needing this variable should it just be in the name of the consolidator type?